### PR TITLE
add db test in read-only mode

### PR DESCRIPTION
**Pull request by lorneli**

### CODE REVIEW

**Comment by lorneli** [comment]
This pr tests read-only mode db, which should increase test coverage. :sweat: https://codecov.io/gh/coreos/bbolt/src/master/db.go#L179

File: db_test.go [hunk 1]
```diff
@@ -1125,6 +1125,61 @@ func TestDB_BatchTime(t *testing.T) {
...
+	err := db.DB.Close()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	f := db.f
+	o := &bolt.Options{ReadOnly: true}
+	readOnlyDB, err := bolt.Open(f, 0666, o)
+	if err != nil {
+		panic(err)
```
Comment (heyitsanthony) [inline db_test.go]: t.Fatal(err)

File: db_test.go [hunk 2]
```diff
@@ -1125,6 +1125,61 @@ func TestDB_BatchTime(t *testing.T) {
...
+
+	// Read from a read-only transaction.
+	if err := readOnlyDB.View(func(tx *bolt.Tx) error {
+		value := tx.Bucket([]byte("widgets")).Get([]byte("foo"))
+		if bytes.Compare(value, []byte("bar")) != 0 {
+			t.Fatal("expect value 'bar', got", value)
+		}
+		return nil
+	}); err != nil {
+		log.Fatal(err)
```
Comment (heyitsanthony) [inline db_test.go]: t.Fatal(err)

File: db_test.go [hunk 3]
```diff
@@ -1125,6 +1125,61 @@ func TestDB_BatchTime(t *testing.T) {
...
+	}
+
+	// Can't launch read-write transaction.
+	if _, err := readOnlyDB.Begin(true); err != bolt.ErrDatabaseReadOnly {
+		t.Fatalf("unexpected error: %s", err)
+	}
+
+	if err := readOnlyDB.Close(); err != nil {
+		log.Fatal(err)
```
Comment (heyitsanthony) [inline db_test.go]: t.Fatal(err)

File: db_test.go [hunk 4]
```diff
@@ -1125,6 +1125,61 @@ func TestDB_BatchTime(t *testing.T) {
 	}
 }
 
+// Ensure that a database in read only mode can read from read-only transaction,
+// but can't launch read-write transaction.
+func TestDB_ReadOnlyMode(t *testing.T) {
```
Comment (heyitsanthony) [inline db_test.go]: `TestDB_Open_ReadOnly` to match `TestDB_Open_InitialMmapSize`

File: db_test.go [hunk 5]
```diff
@@ -1449,15 +1504,7 @@ type DB struct {
 
 // MustOpenDB returns a new, open DB at a temporary location.
 func MustOpenDB() *DB {
```
Comment (heyitsanthony) [inline db_test.go]: squash commit

File: db_test.go [hunk 6]
```diff
@@ -1125,6 +1125,61 @@ func TestDB_BatchTime(t *testing.T) {
...
+			t.Fatal(err)
+		}
+		if err := b.Put([]byte("foo"), []byte("bar")); err != nil {
+			t.Fatal(err)
+		}
+		return nil
+	}); err != nil {
+		t.Fatal(err)
+	}
+	err := db.DB.Close()
```
Comment (heyitsanthony) [inline db_test.go]:
```go
if err := db.DB.Close(); err != nil {
```
to match other if's

File: db_test.go [hunk 7]
```diff
@@ -1125,6 +1125,61 @@ func TestDB_BatchTime(t *testing.T) {
 	}
 }
 
+// Ensure that a database in read only mode can read from read-only transaction,
```
Comment (heyitsanthony) [inline db_test.go]: use godoc style:
```go
TestDB_Open_ReadOnly checks a database in read only mode can read but not write.
```

**Comment by lorneli** [comment]
Update commit.

**Comment by xiang90** [comment]
@lorneli why this is closed?

**Comment by heyitsanthony** [comment]
@lorneli is this OK to merge?

**Comment by xiang90** [comment]
@heyitsanthony I believe he accidentally closed the PR when commenting. we can just merge this in.

**Comment by lorneli** [comment]
@heyitsanthony @xiang90 I accidentally closed this PR when commenting on mobile phone. Its screen is too small.... Thank you for your review.

**Pull request merged**